clash-lib/src/Clash/Normalize/Transformations/Specialize.hs

```diff
@@ -486,6 +487,17 @@ specialize' (TransformContext is0 _) e (Var f, args, ticks) specArgIn = do
     noUserInline = NoUserInline
 #endif
 
+    specializeName :: Name Term -> Name Term -> Name Term
+    specializeName n0 n1 =
+      n1{nameOcc=stripMods (nameOcc n0) <> "_" <> stripMods (nameOcc n1)}
```

Comment:
This throws away provenance info of the first name. While this is not a regression over the status-quo, it's probably still good to add a comment. Perhaps we want to store a tree of provenance info in `Name`s in the future.

Reply:
I think we should remember the provonance of at least one of the names. To make things simpler, that would be the provenance of the function being specialized.

Reply:
Would you agree that in the future it could be desirable to remember both names? (Not in this PR though.)